From d06f9742d1178f738adccd8e2648fbb7e0bd4e5a Mon Sep 17 00:00:00 2001
From: Roald Clark <roaldclark@gmail.com>
Date: Tue, 10 Dec 2024 23:16:57 +0800
Subject: [PATCH] Add 'headers' argument and simplify header handling

- Add 'headers' argument to the setup() function in setup.py
- Remove code for installing header files in virtual environments

Signed-off-by: Roald Clark <roaldclark@gmail.com>
---
 setup.py | 20 +-------------------
 1 file changed, 1 insertion(+), 19 deletions(-)

diff --git a/setup.py b/setup.py
index 63bfaef..5506399 100755
--- a/setup.py
+++ b/setup.py
@@ -150,7 +150,7 @@ if __name__ == "__main__":
     setup(
         "pygame_sdl2",
         VERSION,
-        headers=py_headers,
+        headers=headers,
         url="https://github.com/renpy/pygame_sdl2",
         maintainer="Tom Rothamel",
         maintainer_email="tom@rothamel.us",
@@ -160,21 +160,3 @@ if __name__ == "__main__":
 
     for i in temporary_package_data:
         os.unlink(os.path.join(os.path.dirname(__file__), "src", "pygame_sdl2", i))
-
-    if headers:
-        import pathlib
-
-        virtual_env = os.environ.get("VIRTUAL_ENV", None)
-
-        if virtual_env:
-            headers_dir = pathlib.Path(virtual_env) / "include" / "pygame_sdl2"
-        else:
-            headers_dir = pathlib.Path(sysconfig.get_paths()['include']) / "pygame_sdl2"
-
-        headers_dir.mkdir(parents=True, exist_ok=True)
-
-        for header in headers:
-            srcpath = pathlib.Path(header)
-            dstpath = headers_dir / srcpath.name
-
-            shutil.copy(srcpath, dstpath)
-- 
2.48.1

